src/Entities/UserEntity.php

```diff
@@ -1,3 +1,11 @@
 <?php
 
+declare(strict_types=1);
 
+class UserEntity
+{
+    private int $id;
+    private string $username;
+    private string $email;
+    private string $password;
```

Comment:
It may be worth adding getter methods in here now, even though you currently don't actually need to access any of these properties, you may well need to in future.